pkg/graph/resolvers/system_intake.go

```diff
@@ -0,0 +1,193 @@
+package resolvers
+
+import (
+	"context"
+
+	"github.com/guregu/null"
+
+	"github.com/cmsgov/easi-app/pkg/graph/model"
+	"github.com/cmsgov/easi-app/pkg/graph/resolvers/systemintake/formstate"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// TODO: thes calls could largely be combined in a more general call to Update the System Intake. It would rely on a similar approach that was taken in TRB
```

Comment:
Out of curiosity, did the approach the TRB used handle multiple models/tables? I could see how `SystemIntakeUpdate()` and `SystemIntakeUpdateContactDetails()` might be combined, but `SystemIntakeUpdateContractDetails()` needs to call `store.UpdateSystemIntakeFundingSources()` to update the `system_intake_funding_sources` table as well as updating the intake itself.